### PR TITLE
Migrate debug page to hooks.

### LINKED ISSUE #1949: Convert React class components to React hooks
I find React hooks are much cleaner and less confusing than the old class-based React components. We convert our class-based components to hooks.

### PR DESCRIPTION
**Pull request by anuraaga**
It'd be nice if you can check out the debug page in this branch locally to make sure it works ok - I tried various operations and think it's fine but there are a lot of possible variations.

Also fixes long-time pet peeve by prettifying gRPC unframed JSON.

Fixes #1949

### CODE REVIEW

File: docs-client/src/lib/transports/grpc-unframed.ts [hunk 1]
```diff
@@ -51,7 +52,6 @@ export default class GrpcUnframedTransport extends Transport {
       method: 'POST',
       body: bodyJson,
     });
-    const response = await httpResponse.text();
-    return response;
+    return jsonPrettify(await httpResponse.text());
```
Comment (trustin) [inline docs-client/src/lib/transports/grpc-unframed.ts]: :tada:

**Comment by ikhoon** [comment]
While I was testing I found an exception. I clicked `Submit` button with any input and I saw a black page and exeception message.
```js

react-dom.development.js?61bb:13124 Uncaught TypeError: Method Error.prototype.toString called on incompatible receiver undefined
    at toString (<anonymous>)
    at basicStateReducer (react-dom.development.js?61bb:13124)
    at updateReducer (react-dom.development.js?61bb:13246)
    at updateState (react-dom.development.js?61bb:13294)
    at Object.useState (react-dom.development.js?61bb:13843)
    at useState (react.development.js?72d0:1462)
    at DebugPage (DebugPage.tsx?725f:115)
    at DebugPage (react-hot-loader.development.js?c2cb:821)
    at renderWithHooks (react-dom.development.js?61bb:12938)
    at updateFunctionComponent (react-dom.development.js?61bb:14627)
```
TestSrevice UnaryCall2

**Comment by anuraaga** [comment]
@ikhoon Thanks for finding, very silly bug!

**Comment by ikhoon** [comment]
@anuraaga Your PR is always welcome. I'm happy to find and share it 😉

**Comment by ikhoon** [comment]
I think `404 Bad Request` message doesn't print on debug console.

unary-method

This is master version screen shot.😀
Screen Shot 2019-10-10 at 11 13 54 AM

**Comment by anuraaga** [comment]
Thanks - fixed

**Comment by anuraaga** [comment]
I think both were introduced in previous PRs (by me ><). Fixed the error, and also removed a spacing parameter that was causing the hover issue. The method type / name are a bit closer now, but I guess it still looks ok to me, though I might be forgetting the reason the parameter was there in the first place.

**Pull request merged**